### PR TITLE
Rename `ConcatFilesDataset` to `DistributeFilesDataset`

### LINKED ISSUE #1535: `ConcatFilesDataset` needs a better name
#1521, #1519.

The name is misleading given all the other concatenation-related datasets we have, because people assume that the dataset will somehow modify the data itself when its true purpose is assisting in large-scale trainings. This hinders explainability.

A name focusing on the intent might make more sense to many people, like:
- `LazyLoadingDataset`
- `ShardBySubepochDataset` 
- maybe even `LargeScaleDataset`

Feel free to dump more ideas into this issue. Then we'll decide on a new name by June 11, 5pm CEST.

### PR DESCRIPTION
**Pull request by NeoLegends**
Closes #1535, using Albert's proposal that got two :+1: 

@Icemole this is a breaking change, which we deemed ok because nobody's got a real setup with this dataset yet

### CODE REVIEW

File: returnn/datasets/distrib_files.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """
-:class:`ConcatFilesDataset `
+:class:`DistributeFilesDataset `
```
Comment (albertz) [inline returnn/datasets/distrib_files.py:2]:
```suggestion
:class:`DistributeFilesDataset`
```

File: returnn/datasets/distrib_files.py [hunk 2]
```diff
@@ -23,19 +23,17 @@
...
+class DistributeFilesDataset(CachedDataset2):
     """
-    This is similar to :class:`ConcatDataset`, but instead of concatenating datasets,
-    we distribute files over subepochs,
-    and then create a sub dataset for every sub epoch
-    for a given subset of the files
-    via the given ``get_sub_epoch_dataset``.
+    Wrapper dataset that distributes files over subepochs and then creates a
+    sub dataset for every sub epoch for a given (random) subset of the files.
+    The sub dataset is user-defined via a function ``get_sub_epoch_dataset``.
```
[the remark this answers is not in the document]
Comment (NeoLegends) [reply returnn/datasets/distrib_files.py:36]: I do like mentioning that this is a wrapper dataset though, because it cannot work on its own.

**Comment by Icemole** [comment]
Maybe `Distribute[d]FilesDataset` instead of `DistributeFilesDataset`? Seems more consistent with respect to e.g. `CachedDataset`.

File: returnn/datasets/distrib_files.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (albertz) [reply returnn/datasets/distrib_files.py:36]: Isn't this clear when we say that it creates sub datasets? I'm not sure that calling this a "wrapper dataset" is maybe confusing. We don't use this term on e.g. `MetaDataset`, `ConcatDataset`, `CombinedDataset`.

[the remark this answers is not in the document]
Comment (albertz) [reply returnn/datasets/distrib_files.py:36]: I added one sentence now. But not sure about it.

[the remark this answers is not in the document]
Comment (NeoLegends) [reply returnn/datasets/distrib_files.py]: I like that it's there now -- doesn't need to be high up in the sentence.

**Comment by albertz** [comment]
> Maybe `Distribute[d]FilesDataset` instead of `DistributeFilesDataset`? Seems more consistent with respect to e.g. `CachedDataset`.

Looking through other dataset names, it's not really so consistent. I don't really have any strong opinion here. Both variants are fine.

**Comment by NeoLegends** [comment]
Merge then once CI is green?

**Pull request merged**